### PR TITLE
feat(core): add PATCH /configs/jwt-customizer API

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add `PATCH /configs/jwt-customizer/:tokenType` API

## Testing

Covered by unit tests and integration tests.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/logto-config.ts [hunk 1]
```diff
@@ -39,6 +39,16 @@ const getLogtoJwtTokenKey = (key: LogtoJwtTokenKeyType): LogtoJwtTokenKey =>
     ? LogtoJwtTokenKey.AccessToken
     : LogtoJwtTokenKey.ClientCredentials;
 
+const guardRequestBodyByTokenType = (tokenType: LogtoJwtTokenKeyType, body: unknown) => {
```
Comment (simeng-li) [inline packages/core/src/routes/logto-config.ts:42]: please refer to the previous PR.  Can return a guarded type value instead of using as.

**Comment by darcyYe** [comment]
We merged POST/PATCH API as a single PUT API (see [this PR](https://github.com/logto-io/logto/pull/5461)), close this due to no longer needed.

File: packages/core/src/libraries/logto-config.ts [hunk 2]
```diff
@@ -69,9 +70,12 @@ export const createLogtoConfigLibrary = ({
...
 
   // Can not narrow down the type of value if we utilize `buildInsertIntoWithPool` method.
-  const upsertJwtCustomizer = async <T extends LogtoJwtTokenKey>(
+  const upsertJwtCustomizer = async <
+    T extends LogtoJwtTokenKey,
+    U extends z.infer<(typeof jwtCustomizerConfigGuard)[T]>,
+  >(
     key: T,
-    value: z.infer<(typeof jwtCustomizerConfigGuard)[T]>
+    value: U
```
Comment (simeng-li) [inline packages/core/src/libraries/logto-config.ts:78]: the second generic type U is unnecessary? 

```
value: z.infer<(typeof jwtCustomizerConfigGuard)[T]>
```

File: packages/core/src/libraries/logto-config.ts [hunk 3]
```diff
@@ -97,5 +101,28 @@ export const createLogtoConfigLibrary = ({
...
+  /**
+   * We only guard the type of `key` field here.
+   *
+   * The guard of `value` field is handle on:
+   * 1. API request interface.
+   * 2. Manually run zod guard after merging the `value` and the existing record.
+   */
+  const updateJwtCustomizer = async <T extends LogtoJwtTokenKey>(
+    key: T,
+    value: Record<string, unknown>
```
Comment (simeng-li) [inline packages/core/src/libraries/logto-config.ts:113]: should we guard the value here? e.g. Partial<z.infer<(typeof jwtCustomizerConfigGuard)[T]>

**Review by wangsijie — APPROVED** [review]

**Pull request merged**